frameworks/kotlin/build.gradle

```diff
@@ -0,0 +1,75 @@
+buildscript {
+  ext.kotlin_version = '1.1.3'
+  repositories {
+    jcenter()
+  }
+  dependencies {
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+  }
+}
+
+apply plugin: 'kotlin'
+apply plugin: 'application'
+
+mainClassName = System.getenv('KOTLIN_MAIN_CLASS_NAME')
+defaultTasks 'run'
+
+repositories {
+  jcenter()
+}
+
+dependencies {
+  compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+  testCompile 'junit:junit:4.12'
+  testCompile "org.jetbrains.kotlin:kotlin-test-junit:$kotlin_version"
+}
+
+compileKotlin {
+  kotlinOptions {
+  }
+}
+
+test {
+  reports {
+    html.enabled = false
+    junitXml.enabled = false
+  }
+
+  beforeSuite { desc ->
```

Comment:
OK, we can keep it separate for now. I also wanted to update the java version to use listeners at some point. I think the only thing that is special to the CwRunListener is the `<LOG::-Exception Details>` bit, which probably also be doable within the listeners.